app.js

```diff
@@ -34,62 +29,61 @@ let allProducts = [
   new Product('wine-glass'),
 ];
 
+//--------------------CONSTRUCTORS
+
+// Product Constructor:
+function Product(name) {
+  this.name = name;
+  this.source = `img/${name}.jpg`;
+  this.viewed = 0;
+  this.clicked = 0;
+}
+
+// constructor to capitalize first letter of strings
+String.prototype.capitalize = function () {
+  return this.charAt(0).toUpperCase() + this.slice(1);
+};
 
-// // Instead of a for loop, I want to use a random number generator
-// for (let i = 0; i < allProducts.length; i++) {
-// let product = allProducts[i];
-// // Step 1. create element
-// // - Done in html, line 11
-// // Step 2. fill with content
-// let img = document.getElementById("productImage");
-// console.log(img);
-// img.src = `assets/${product.name}.jpg`
-// // Step 3. append child
-// }
+
+//--------------------CONSTRUCTOR METHODS
+
+//--------------------FUNCTIONS
 
 // return a random index inside allProducts array
 function randomImage() {
   // return a number between 0 - 6 (6 is the index of the last item in allProducts)
   return Math.floor(Math.random() * allProducts.length);
 }
 
-// let product = allProducts[randomImage()];
-
-
-// Generate 3 unique indexes
-
-let imageArray = [];
-
+// Generate 19 unique indexes
 function uniqueImageChecker () {
   // while there are fewer than 3 images
-  while (imageArray.length < 3) {
+  while (imageArray.length < 19) {
```

Comment:
This number 19 is making your images not random. The goal was to make sure images were unique between any two sets of three, but this causes your images to cycle through the array randomly, which makes your later sets not random.